File: leetcode/algorithms/1.TwoSum/Solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var hash = {};
    for(var i=0; i<nums.length; i++) {
        hash[nums[i]] = i;
    }
    
    for(var i=0; i<nums.length; i++) {
        var needValue = target - nums[i];
        if(hash.hasOwnProperty(needValue)) {
            var index1 = i+1;
            var index2 = hash[needValue]+1;
            if(index1!==index2) {
                return [index1, index2];
            }
        }
    }
};

File: leetcode/algorithms/1.TwoSum/Solution.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, nums, target):
        lookup = {}
        for i, num in enumerate(nums):
            if target - num in lookup:
                return (lookup[target - num] + 1, i + 1)
            lookup[num] = i
